insights/util/canonical_facts.py

```diff
@@ -123,22 +124,27 @@ def _filter_falsy(dict_):
         Specs.bios_uuid,
         SubscriptionManagerID,
         IPs,
+        AWSPublicIPv4,
         Specs.hostname,
         Specs.mac_addresses,
         CloudInstance,
     ]
 )
 def canonical_facts(
-    insights_id, machine_id, bios_uuid, submanid, ips, fqdn, mac_addresses,
+    insights_id, machine_id, bios_uuid, submanid, ips, aws_ipv4, fqdn, mac_addresses,
     cloud_instance,
 ):
 
+    ips_list = ips.data if ips else []
+    if aws_ipv4:
+        ips_list.append(aws_ipv4.data)
+
     facts = dict(
         insights_id=valid_uuid_or_None(_safe_parse(insights_id)),
         machine_id=valid_uuid_or_None(_safe_parse(machine_id)),
         bios_uuid=valid_uuid_or_None(_safe_parse(bios_uuid)),
         subscription_manager_id=valid_uuid_or_None(submanid.data if submanid else None),
-        ip_addresses=ips.data if ips else [],
+        ip_addresses=ips_list,
```

Comment:
The canonical facts JSON schema is not (to my knowledge) strictly defined anywhere, but it is assumed to include a certain set of keys (in other words, it follows a de facto standard). There are other implementations in code that (for instance, [rhc](https://github.com/RedHatInsights/rhc/blob/main/canonical_facts.go#L112) and in [cloud-connector](https://github.com/RedHatInsights/cloud-connector/blob/90a7ec8ae8cc5a201a0fd05f9afd4f565ced4a9a/internal/cloud_connector/protocol/types.go#L42-L51)) that assume the current schema is adhered to. I suggest you take care when making changes to canonical facts. This change is safe, since you don't appear to be altering the key strings at all, but the above history of canonical facts is important to keep in mind.

Reply:
FYI, here's the list of canonical facts: https://consoledot.pages.redhat.com/docs/dev/services/inventory.html#_host_deduplication